pkg/ingester/limiter.go

```diff
@@ -33,6 +33,10 @@ type Limits interface {
 	PerStreamRateLimit(userID string) validation.RateLimit
 	ShardStreams(userID string) shardstreams.Config
 	IngestionPartitionsTenantShardSize(userID string) int
+	RetentionPeriod(userID string) time.Duration
```

Comment:
You can add `retention.Limits` here instead
```go
type Limits interface {
	...
	ShardStreams(userID string) shardstreams.Config
	IngestionPartitionsTenantShardSize(userID string) int
	retention.Limits
}
```

Reply:
hah good catch :smile: